app/src/app/components/specmath-modal/modal.component.ts

```diff
@@ -57,8 +50,36 @@ export class ModalComponent implements OnInit {
     }
   }
 
-  get specNameValid(): boolean {
-    return !!this.specNameInputOptions?.valid;
+  get newFileName(): string {
+    if (!this.specNameInputOptions) {
+      return '';
+    }
+
+    return this.specNameInputOptions.newFileName;
+  }
+
+  get nextButtonTooltipText(): string {
+    if (this.currentStep === 1) {
+      return 'You must name your new spec';
+    }
+
+    if (this.currentStep === 3) {
+      return 'You must upload a set of spec files';
+    }
+
+    return '';
+  }
+
+  get nextButtonDisabled(): boolean {
+    if (this.currentStep === 1 && !!!this.specNameInputOptions?.valid) {
```

Comment:
Just curious, why !!! here rather than just ! ?

Reply:
Since `this.specNameInputOptions` might not be defined, the `!!` turns it into false if its undefined, adding the extra `!` just negates whatever the value is. I'm open to suggestions tho since it looks a bit confusing lol

Reply:
Changed to use only one `!` found out using three ends up with the same outcome.